README.md

```diff
@@ -58,26 +58,35 @@ contract('ERC20', function ([sender, receiver]) {
 
 ### Configuration
 
-This library supports both web3 and truffle contract instances. Where possible, helpers will automatically detect what you're using and work with both. For details about each helper see the documentation below.
+This library supports both web3 and truffle contract instances. Where possible, helpers will automatically detect what you're using and work with both. For details about each helper see their documentation entries below.
 
-In a truffle environment, the web3 provider will be pulled from truffle's global web3 instance. Otherwise, it defaults to `http://localhost:8545`.
+#### provider
 
-The contract abstraction type `singletons` returns depends on the `contractAbstraction` configuration value. The default one is `'web3'`, unless a `truffle` environment is automatically detected.
-
-While automatic detection should cover most use cases, both provider and contract abstraction can be manually supplied:
+In a truffle environment, the web3 provider will be pulled from truffle's global web3 instance. Otherwise, it defaults to `http://localhost:8545`. You can override this behavior and configure your own via the `provider` key:
 
 ```javascript
-require('@openzeppelin/test-helpers/configure')({ contractAbstraction: 'web3', provider: 'http://localhost:8080' });
+require('@openzeppelin/test-helpers/configure')({ provider: 'http://localhost:8080' });
+````
+
+#### singletons
 
-const { expectEvent } = require('openzeppelin-test-helpers');
+The `singletons` helper returns contract objects, which have multiple values that can be configured:
+ * `abstraction`: the underlying contract abstraction type, `'web3'` for `web3-eth-contract` and `'truffle'` for `@truffle/contract` instances. Defaults to `'web3'` unless a truffle environment is detected.
+ * `defaultGas`: how much gas to allocate when a transaction's `gas` field is not specified. Defaults to 8 million.
```

Comment:
The 8 million default is waaay too high. Absolutely too high. Dangerously too high. Even if test-helpers is meant to run in testing environments we don't know if someone will use it in mainnet.

Besides, this will probably not work in any non-Ganache blockchain because it's so close or above the block gas limit.

Reply:
I could've sworn that ganache's block limit was 8 million, but turns out it is instead 6721975. What default value do you suggest?